assets/pages/new-contributions-landing/components/ContributionForm.jsx

```diff
@@ -106,19 +117,43 @@ const mapDispatchToProps = (dispatch: Function) => ({
   updateLastName: (event) => { dispatch(updateLastName(event.target.value)); },
   updateEmail: (event) => { dispatch(updateEmail(event.target.value)); },
   updateState: (event) => { dispatch(updateState(event.target.value === '' ? null : event.target.value)); },
-  onWaiting: (isWaiting) => { dispatch(paymentWaiting(isWaiting)); },
+  setPaymentIsWaiting: (isWaiting) => { dispatch(paymentWaiting(isWaiting)); },
   onThirdPartyPaymentAuthorised: (token) => { dispatch(onThirdPartyPaymentAuthorised(token)); },
   setCheckoutFormHasBeenSubmitted: () => { dispatch(setCheckoutFormHasBeenSubmitted()); },
   openDirectDebitPopUp: () => { dispatch(openDirectDebitPopUp()); },
+  // Not intended to be called by component.
+  // Only used to implement createOneOffPayPalPayment in mergeProps()
+  _createOneOffPayPalPayment: (data: CreatePaypalPaymentData) => { dispatch(createOneOffPayPalPayment(data)); },
 });
 
-// ----- Functions ----- //
 
-const getAmount = (props: PropTypes) =>
+// TODO: some type-safety
+const getAmount = (props: Object) =>
   parseFloat(props.selectedAmounts[props.contributionType] === 'other'
     ? props.otherAmount
     : props.selectedAmounts[props.contributionType].value);
 
+
+const mergeProps = (stateProps, dispatchProps, ownProps) => ({
+  ...stateProps,
+  ...dispatchProps,
+  ...ownProps,
+  // TODO: is it a good idea putting this function here?
+  // Allows the component to create a payment without having to think directly about what arguments are required.
+  createOneOffPayPalPayment: () => {
```

Comment:
Re. the conversation between @joelochlann, @Mullefa and I, I've fallen on the side of not having mergeProps and doing this in the component. I think it looks a bit messy, as seem by this comment above:

`// Not intended to be called by component.
  // Only used to implement createOneOffPayPalPayment in mergeProps()`

Reply:
I agree.

Reply:
me too! fixed